src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/FeatureFlag/FeatureFlagsFormDataProvider.php

```diff
@@ -28,42 +28,91 @@
+use Doctrine\ORM\EntityManagerInterface;
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
 use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use PrestaShopBundle\Entity\FeatureFlag;
 
 /**
  * Passes data between the application layer in charge of the feature flags form
  * and the domain layer in charge of the feature flags model
  */
 class FeatureFlagsFormDataProvider implements FormDataProviderInterface
 {
-    /**
-     * @var DataConfigurationInterface
-     */
-    private $featureFlagDataConfigurator;
+    /** @var EntityManagerInterface */
+    protected $doctrineEntityManager;
+
+    /** @var string */
+    protected $stability;
 
     /**
-     * @param DataConfigurationInterface $featureFlagDataConfigurator
+     * @param EntityManagerInterface $doctrineEntityManager
+     * @param string $stability
      */
-    public function __construct(
-        DataConfigurationInterface $featureFlagDataConfigurator
-    ) {
-        $this->featureFlagDataConfigurator = $featureFlagDataConfigurator;
+    public function __construct(EntityManagerInterface $doctrineEntityManager, string $stability)
+    {
+        $this->doctrineEntityManager = $doctrineEntityManager;
+        $this->stability = $stability;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getData(): array
+    public function getData()
     {
-        return $this->featureFlagDataConfigurator->getConfiguration();
+        $featureFlags = $this->doctrineEntityManager->getRepository(FeatureFlag::class)->findBy(['stability' => $this->stability]);
+
+        $featureFlagsData = [];
+        foreach ($featureFlags as $featureFlag) {
+            $featureFlagsData[$featureFlag->getName()] = [
+                'enabled' => $featureFlag->isEnabled(),
+                'name' => $featureFlag->getName(),
+                'label' => $featureFlag->getLabelWording(),
+                'label_domain' => $featureFlag->getLabelDomain(),
+                'description' => $featureFlag->getDescriptionWording(),
+                'description_domain' => $featureFlag->getDescriptionDomain(),
+            ];
+        }
+
+        return $featureFlagsData;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function setData(array $data): array
+    public function setData(array $flagsData)
+    {
+        if (!$this->validateFlagsData($flagsData)) {
+            throw new InvalidArgumentException('Invalid feature flag configuration submitted');
+        }
+
+        foreach ($flagsData as $flagName => $flagData) {
+            $featureFlag = $this->getOneFeatureFlagByName($flagName);
+
+            if (null === $featureFlag) {
+                throw new InvalidArgumentException(sprintf('Invalid feature flag configuration submitted, flag %s does not exist', $flagName));
+            }
+
+            $flagState = $flagData['enabled'] ?? false;
+            $flagState ? $featureFlag->enable() : $featureFlag->disable();
```

Comment:
@Progi1984 you realize using a ternary here only reduces code readability since it's not a variable association nor a return or did you let your bot review for you again?